src/System.Buffers.Primitives/System/Buffers/OwnedBuffer.cs

```diff
@@ -69,7 +40,11 @@ public void Dispose()
         public abstract void Retain();
 
         public abstract void Release();
-
         #endregion
+
+        protected static unsafe void* Add(void* pointer, int offset)
```

Comment:
This is code that implementers of OwnedBuffers will have to write. I would prefer this be simpler than going back and forth between references and pointers. Let me leave the PR as-is and we can think separately what's the best way to provide convenient API for this scenario. I filed issue #1529